### ISSUE TITLE
The engine can miss early win ; case at depth 4

### ISSUE THREAD

**Issue by LucasBorboleta** [eclypse-prime/pijersi-rs#17]
In the following game, Natural-Selection at depth 4 (whites) plays against Cmalo at depth 4 (blacks). At action 37 (pijersi-certu notation), blacks have two cubes (rock and wise), and whites could win. But whites continues until action 54 (pijersi-certu notation) and terminates the game with a draw. Note : the setup is half-random.

-----

_Nobody wins ; the game is a draw between white-cmalo-depth-4 and black-natsel-depth-4 ; 4171 versus 30 seconds_

   ```
g16:psrsws f13:wps f4:rp f57:rpr
   b13:RPR b4:RP b57:SPW a16:SWSRSP

   1 a1-b2=d1             2 f1-e1           
   3 b4=d5                4 g3-f3=d2        
   5 d1=b2-c1             6 g1-f2=d3        
   7 a5-a4                8 d3=d5!-e4       
   9 b2-b3=d2!           10 f4=f3-f2        
  11 d2=e2-f2!           12 g2-f2!          
  13 e2-f2!              14 f3-f2!          
  15 c1-b1=d2            16 f5-e4=e2        
  17 a4-b5=d4            18 e2=f3           
  19 d2=e2-f2!           20 f3=f2!-f3       
  21 d4=d5!              22 f7-f6=f4        
  23 d5=d3-e3            24 f4=e3!-d3!      
  25 a6-b6=d5            26 g4-f4           
  27 a3-a2=b3            28 f2-e2!          
  29 b3=d4               30 d3-d4!          
  31 d5=d6               32 f4-e3=e5        
  33 d6=f7               34 g6-f7!          
  35 b7-b6               36 e5=c4-d4        
  37 b6-c6               38 g5-g4           
  39 a4-b4               40 g4-g3           
  41 b4-a4               42 g3-g2           
  43 c6-d6               44 g2-g1           
  45 d6-e5               46 g1-g2           
  47 a4-a3               48 g2-g1           
  49 e5-f5               50 g1-g2           
  51 f5-f4               52 g2-g1           
  53 f4-g3               54 g1-g2           


```

**Comment by LucasBorboleta** [comment]
Same game but with UFI FEN position and answer of Natural Selection starting from move 37 ; see attachement ;
in short : 
FEN position : ['4w-1/2r-3s-/w-p-4/3rs3/3p-1W-/7/3R-2', 'b', '3', '19']
Natural-Selection answers : g5g4
So Natural-Selection is missing a winning move
[debug-game-cmalo-natsel-with-UGI-FEN-positions.txt](https://github.com/user-attachments/files/16329378/debug-game-cmalo-natsel-with-UGI-FEN-positions.txt)

**Comment by eclypse-prime** [comment]
This was fixed by #9 it will be part of release 1.0.0
